Realm/RLMConstants.h

```diff
@@ -18,6 +18,14 @@
 
 #import <Foundation/Foundation.h>
 
+// For compatibility with Xcode 7, before extensible string enums were introduced,
+#ifdef NS_EXTENSIBLE_STRING_ENUM
+#define RLM_EXTENSIBLE_STRING_ENUM_CASE_SWIFT_NAME(_, extensible_string_enum) NS_SWIFT_NAME(extensible_string_enum)
+#else
+#define NS_EXTENSIBLE_STRING_ENUM
```

Comment:
@bdash The macro needs to be used in our public headers though (since `NS_EXTENSIBLE_STRING_ENUM` and `NS_SWIFT_NAME` annotations must be in the header). Do you think that the `RLM` prefix isn't sufficient?